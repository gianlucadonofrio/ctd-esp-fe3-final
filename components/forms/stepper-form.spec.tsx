import { render, screen } from "@testing-library/react";
import { comic } from "dh-marvel/test/mocks/comic";
import { FormProvider, useForm } from "react-hook-form";
import { FC, PropsWithChildren } from "react";
import { ICheckout } from "types/ICheckout.type";
import StepperForm from "./stepper-form.component";

describe("StepperForm component", () => {
  const steps = ["Datos Personales", "Direcci√≥n de entrega", "Datos del pago"];

  const Wrapper: FC<PropsWithChildren<{}>> = ({ children }) => {
    const methods = useForm<ICheckout>({
      defaultValues: {
        personalData: {
          nombre: "",
          apellido: "",
          email: "",
          direccion: {
            calle: "",
            ciudad: "",
            provincia: "",
            codigoPostal: "",
          },
        },
        paymentData: {
          cvc: "",
          expDate: "",
          nameOnCard: "",
          number: "",
        },
        orderData: {
          nombre: "",
          cantidad: 0,
          total: 0,
          imagen: "",
        },
      },
    });

    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  it("should render the component", () => {
    render(<StepperForm comic={comic} />, { wrapper: Wrapper });

    steps.forEach((step) => {
      expect(screen.getByText(step)).toBeInTheDocument();
    });
  });

  
});
