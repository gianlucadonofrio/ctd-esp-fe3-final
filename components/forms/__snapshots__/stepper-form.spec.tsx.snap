// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepperForm should render the component 1`] = `
<div>
  <div
    class="MuiBox-root css-8atqhb"
  >
    <div
      class="MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel css-10mg1vw-MuiStepper-root"
    >
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root"
      >
        <button
          class="MuiButtonBase-root MuiStepButton-root MuiStepButton-horizontal css-mjpk2e-MuiButtonBase-root-MuiStepButton-root"
          color="inherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer Mui-active MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  1
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="MuiStepLabel-label Mui-active MuiStepLabel-alternativeLabel css-qivjh0-MuiStepLabel-label"
              >
                Datos Personales
              </span>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root MuiStepButton-touchRipple css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root"
      >
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <button
          class="MuiButtonBase-root Mui-disabled MuiStepButton-root MuiStepButton-horizontal css-mjpk2e-MuiButtonBase-root-MuiStepButton-root"
          color="inherit"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  2
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-qivjh0-MuiStepLabel-label"
              >
                Direcci√≥n de entrega
              </span>
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel css-n7tliy-MuiStep-root"
      >
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled css-zpcwqm-MuiStepConnector-root"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
          />
        </div>
        <button
          class="MuiButtonBase-root Mui-disabled MuiStepButton-root MuiStepButton-horizontal css-mjpk2e-MuiButtonBase-root-MuiStepButton-root"
          color="inherit"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel css-ascpo7-MuiStepLabel-root"
          >
            <span
              class="MuiStepLabel-iconContainer Mui-disabled MuiStepLabel-alternativeLabel css-vnkopk-MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1u4zpwo-MuiSvgIcon-root-MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text css-117w1su-MuiStepIcon-text"
                  dominant-baseline="central"
                  text-anchor="middle"
                  x="12"
                  y="12"
                >
                  3
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
            >
              <span
                class="MuiStepLabel-label Mui-disabled MuiStepLabel-alternativeLabel css-qivjh0-MuiStepLabel-label"
              >
                Datos del pago
              </span>
            </span>
          </span>
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="css-nen11g-MuiStack-root"
      >
        <form>
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <div
              class="MuiBox-root css-drqzcc"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                data-testid="nombre-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r0:"
                  id=":r0:-label"
                >
                  Nombre
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    maxlength="20"
                    name="nombre"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Nombre
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <div
              class="MuiBox-root css-drqzcc"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                data-testid="apellido-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r1:"
                  id=":r1:-label"
                >
                  Apellido
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r1:"
                    maxlength="20"
                    name="apellido"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Apellido
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1qm1lh"
          >
            <div
              class="MuiBox-root css-drqzcc"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                data-testid="email-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for=":r2:"
                  id=":r2:-label"
                >
                  Email
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r2:"
                    maxlength="50"
                    name="email"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-1ftyaf0"
                    >
                      <span>
                        Email
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </form>
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiBox-root css-1xhj18k"
          >
            <button
              class="MuiButtonBase-root Mui-disabled MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-mu0rit-MuiButtonBase-root-MuiButton-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              Anterior
            </button>
            <div
              class="MuiBox-root css-i9gxme"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-16irvu6-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Siguiente
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
